sklego/model_selection.py

```diff
@@ -643,3 +643,105 @@ def _regroup(self, groups):
         # create a mapper to set every group to the right group_id
         mapper = dict(zip(df["index"], df["group"]))
         return np.vectorize(mapper.get)(groups)
+
+
+class WithoutLiersCV:
+    """A custom cross-validation splitter that filters out data points labeled as anomalies during the splitting process
+    to exclude them from the training sets.
+
+    The anomaly label is specified by the `anomalous_label` parameter. Data points with this label will be excluded
+    during the splitting process from the training sets, but will be included in the test sets.
+
+    Parameters
+    ----------
+    cv : CV Splitter instance
+        The base cross-validation splitter used to split the data. It must have a `split` method that returns
+        a generator of (train_index, test_index) tuples.
+    anomalous_label : int, default=-1
+        The label used to identify anomalous data points in the target labels.
+        Data points with this label will be excluded during the splitting process from the training sets.
+
+    Example
+    -------
+    ```py
+    import numpy as np
+    from sklearn.model_selection import KFold
+    from sklego.model_selection import WithoutLiersCV
+
+    np.random.seed(1)
+    X = np.random.randn(100, 3)
+    y = np.random.randn(100) > 1.5  # 7% of the data is labeled as anomalous
+
+    cv = WithoutLiersCV(
+        cv=KFold(n_splits=3),
+        anomalous_label=1
+    )
```

Comment:
I think I'd want @MBrouns to weight in on the name 😅 just to make sure. 

But I'm also wondering if it's perhaps easier to the enduser to not require an anomalous label ... wouldn't it perhaps be better to pass in an outlier model? this outlier model could then internally train on X and determine which items are outliers. Or am I overthinking?

Reply:
From the conversation in the issue my understanding is slightly different. The goal of the CV is to validate anomaly detectors that do not train with different labels, namely the novelty detection ones. Therefore passing a novelty detection model would not be possible in the first place.

Now I agree that the name would suit both implementations 😁

Reply:
@koaning Potentially we could have two CV strategies:
- `WithoutLiersCV`: takes _any_ outlier detection model, train on `X`, and excludes outliers from `train_indexes`
- `NoveltyDetectorCV`: what's in this PR, to be able to train a novelty detection algorithm on non-anomalous labels and evaluate on both anomalous and not.